monkey/tests/unit_tests/monkey_island/cc/services/test_agent_signals_service.py

```diff
@@ -117,3 +117,28 @@ def test_progenitor_started_before_terminate(
     signals = agent_signals_service.get_signals(agent.id)
 
     assert signals.terminate.timestamp() == TERMINATE_TIMESTAMP
+
+
+def test_on_terminate_agents_signal__stores_timestamp(
+    agent_signals_service: AgentSignalsService, mock_simulation_repository: ISimulationRepository
+):
+    timestamp = 100
+    mock_simulation_repository.get_simulation = MagicMock(return_value=Simulation())
+    agent_signals_service.on_terminate_agents_signal(timestamp)
+
+    expected_value = Simulation(terminate_signal_time=timestamp)
+    assert mock_simulation_repository.save_simulation.called_once_with(expected_value)
+
+
+def test_on_terminate_agents_signal__updates_timestamp(
```

Comment:
Is it worth adding a test that verifies that the `AgentSignalsService` doesn't mangle the mode (leaves the rest of the simulation untouched)?

Reply:
I think it is as the main thing we want is to update only the timestamp.